at_client/lib/src/util/at_client_validation.dart

```diff
@@ -62,10 +62,32 @@ class AtClientValidation {
     validateMetadata(atKey.metadata);
     // verifies if the sharedWith atSign exists.
     if (atKey.sharedWith != null && await NetworkUtil.isNetworkAvailable()) {
-      isAtSignExists(
+      await isAtSignExists(
```

Comment:
The validity of an @sign needs to be cached.

Reply:
Also while notifying we are verifying "await AtClientUtil.findSecondary(atSign, rootDomain, rootPort);" This needs be cached too.